### DISCUSSION TITLE
What java version does airflow use under the cover?

### DISCUSSION THREAD

**Discussion by vivek-rao-1985** [apache/airflow#26924] [Q&A]
Hi there!
I'm using the ```BeamRunJavaPipelineOperator``` to download a jar containing my beam job and run it on dataflow. When I start my DAG I see this error - ```has been compiled by a more recent version of the Java Runtime```. I'm using java 17 and as far as I understand, the worker node has an older version of java installed but I can't find any documentation on what versions are supported. So what version does airflow use and how can I upgrade it?

**Comment by potiuk** [comment]
Airlfow is pure Python. Java is not installed by default by Airflow. It's part of your deployment, you need to check with whoever instaled Airflow for you.
